content/concepts/hot-module-replacement.md

```diff
@@ -6,52 +6,87 @@ contributors:
 
-In most cases it's not mandatory to write HMR code in every module. If a module has no HMR handlers the update bubbles up. This means a single handler can handle an update to a complete module tree. If a single module in this tree is updated, the complete module tree is reloaded (only reloaded not transferred).
+Similarly, when implementing the HMR interface in a module, you can describe what should
+happen when the module is updated. However, in most cases, it's not mandatory to write
+HMR code in every module. If a module has no HMR handlers, the update bubbles up. This
+means that a single handler can handle an update to a complete module tree. If a single
+module in this tree is updated, the complete module tree is reloaded (only reloaded,
+not transferred).
 
 ### From The HMR Runtime View (Technical)
 
-For the module system runtime additional code is emitted to track module `parents` and `children`.
-
-On the management side the runtime supports two methods: `check` and `apply`.
+For the module system runtime, additional code is emitted to track module `parents` and `children`.
 
-A `check` does a HTTP request to the update manifest. When this request fails, there is no update available. Elsewise the list of updated chunks is compared to the list of currently loaded chunks. For each loaded chunk the corresponding update chunk is downloaded. All module updates as stored in the runtime as update. The runtime switches into the `ready` state, meaning an update has been downloaded and is ready to be applied.
+On the management side, the runtime supports two methods: `check` and `apply`.
 
-For each new chunk request in the ready state the update chunk is also downloaded.
+A `check` makes an HTTP request to the update manifest. If this request fails,
+there is no update available. If it succeeds, the list of updated chunks is compared
+to the list of currently loaded chunks. For each loaded chunk, the corresponding
+update chunk is downloaded. All module updates are stored in the runtime.
+When all update chunks have been downloaded and are ready to be applied, the runtime
+switches into the `ready` state.
 
-The `apply` method flags all updated modules as invalid. For each invalid module there need to be a update handler in the module or update handlers in every parent. Else the invalid buddles up and marks all parents as invalid too. This process continues until no more "bubble up" occurs. If it bubbles up from an entry point the process fails.
+The `apply` method flags all updated modules as invalid. For each invalid module,
+there needs to be an update handler in the module or update handlers in its parent(s).
+Otherwise, the invalid flag bubbles up and marks its parent(s) as invalid too. Each bubble
+continues until the app's entry point or a module with an update handler is reached
+(whichever comes first). If it bubbles up from an entry point, the process fails.
```

Comment:
I am unsure about this paragraph. The existing documentation confused me a bit, so I can rewrite this if it's incorrect.